global_costmap:
    global_frame: "map"
    robot_base_frame: "base_link"
    update_frequency: 3.0
    transform_tolerance: 0.5
    resolution: 0.05
    width: 10.0  # Increased for better global planning
    height: 10.0
    #origin_x: -10.0
    #origin_y: -10.0

    plugins:
        - {name: static_layer, type: "costmap_2d::StaticLayer"}
        - {name: obstacle_layer, type: "costmap_2d::ObstacleLayer"}
        - {name: inflation_layer, type: "costmap_2d::InflationLayer"}

    static_layer:
        enabled: true
        map_topic: /map

    obstacle_layer:
        enabled: true
        observation_sources: lidar_sensor
        footprint_clearing_enabled: true
        lidar_sensor: 
            data_type: LaserScan
            sensor_frame: laser
            topic: /scan
            marking: true
            clearing: true

    inflation_layer:
        enabled: true
        cost_scaling_factor: 10.0  
        inflation_radius: 0.1
